tests/new_api/test_device.py

```diff
@@ -448,6 +465,11 @@ def test_execute_single_measurement(self, theta, phi, mp, dev):
         if isinstance(mp.obs, qml.ops.LinearCombination) and not qml.operation.active_new_opmath():
             mp.obs = qml.operation.convert_to_legacy_H(mp.obs)
 
+        if isinstance(mp.obs, qml.SparseHamiltonian) and dev.dtype == np.complex64:
+            pytest.skip(
+                reason="The conversion from qml.Hamiltonian to SparseHamiltonian is only possible with np.complex128"
```

Comment:
Limitation from pennylane?

Reply:
> https://github.com/PennyLaneAI/pennylane-lightning/pull/908#discussion_r1759155951
> This limitation comes from a hard code in Pennylane [ref](https://github.com/PennyLaneAI/pennylane/blob/060bb9a7479dcf6735671d809da19d500c4b7a89/pennylane/ops/qubit/hamiltonian.py#L522)

Yes, because they set the `dtype` for the sparse matrix conversion with hard code to `complex128`